CRM/Mailing/Info.php

```diff
@@ -92,33 +92,31 @@ public function getAngularModules() {
       ),
     );
 
-    $config = CRM_Core_Config::singleton();
     $session = CRM_Core_Session::singleton();
     $contactID = $session->get('userID');
-    $civiMails = civicrm_api3('Mailing', 'get', array());
-    $campNames = civicrm_api3('Campaign', 'get', array());
-    $mailingabNames = civicrm_api3('MailingAB', 'get', array());
-    $mailStatus = civicrm_api3('MailingJob', 'get', array());
-    $groupNames = civicrm_api3('Group', 'get', array());
-    $headerfooterList = civicrm_api3('MailingComponent', 'get', array());
+
+    $params = array('options' => array('limit' => 0));
+    $civiMails = civicrm_api3('Mailing', 'get', $params);
```

Comment:
There isn't an Angular binding for entityRef, and I agree that it would be useful generally, but the UX/design of that particular field is pretty different.  (EntityRef is at most a single multi-select field; crmMailingRecipients is a synthesis of four multi-select fields.)

Good point that old UI wasn't generating complaints about scalability. The new code is currently a bit worse, though, because it fetches body_html+body_text for all mailings (which is probably 2x-10x bytes). Tweaking it to only return id+name+scheduled_date would be more on-par with the old UI.